### ISSUE TITLE
keep package build times as part of metadata in .spack dir

### ISSUE THREAD

**Issue by cyrush** [LLNL/spack#21179]
### Rationale

During a build, spack provides summary of the build times, like:

```
==> Successfully installed cmake
  Fetch: 1.18s.  Build: 58.42s.  Total: 59.60s.
```
It would be great if these times were also stored along with other info (spack-build-env.txt spack-build-out.txt	spec.yaml) that is kept in the  special `.spack` dir created during the install.

This would allow folks to traverse the builds and extract this info easily after the fact.

### General information

~~- [ ] I have run `spack --version` and reported the version of Spack~~
- [x] I have searched the issues of this repo and believe this is not a duplicate

**Comment by cyrush** [comment]
Here is here the total time message is created:

https://github.com/spack/spack/blob/a8ccb8e116a1abfaa1d7ecb4e84f9f14ebab12ad/lib/spack/spack/installer.py#L1723

The fetch time is calced by the package class, and the total time is calced by the installer. build_process method.  Build time is inferred from total minus fetch, and isn't stored -- just printed. 

The package class provides logic for writing out (copying) the other metadata files,  but given how things are split (installer setting private vars in package), not sure sure where this would go.  Happy to take a pass at it with some guidance.

**Comment by haampie** [comment]
This was fixed by #24350
